File: rust/src/models/plan_event_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{plan_id::PlanId, plan_status_enum::PlanStatusEnum, plan_type_enum::PlanTypeEnum};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PlanEventData {
    pub created_at: String,

    pub currency: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    pub name: String,

    pub plan_id: PlanId,

    pub plan_type: PlanTypeEnum,

    pub status: PlanStatusEnum,

    pub version: i32,
}

impl PlanEventData {
    pub fn new(
        created_at: String,
        currency: String,
        name: String,
        plan_id: PlanId,
        plan_type: PlanTypeEnum,
        status: PlanStatusEnum,
        version: i32,
    ) -> Self {
        Self {
            created_at,
            currency,
            description: None,
            name,
            plan_id,
            plan_type,
            status,
            version,
        }
    }
}

File: typescript/src/models/planEventData.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type PlanId, PlanIdSerializer } from "./planId";
import { type PlanStatusEnum, PlanStatusEnumSerializer } from "./planStatusEnum";
import { type PlanTypeEnum, PlanTypeEnumSerializer } from "./planTypeEnum";

export interface PlanEventData {
  createdAt: Date;

  currency: string;

  description?: string | null;

  name: string;

  planId: PlanId;

  planType: PlanTypeEnum;

  status: PlanStatusEnum;

  version: number;
}

export const PlanEventDataSerializer = {
  _fromJsonObject(object: any): PlanEventData {
    return {
      createdAt: parseDateTime(object["created_at"]),
      currency: object["currency"],
      description: object["description"],
      name: object["name"],
      planId: PlanIdSerializer._fromJsonObject(object["plan_id"]),
      planType: PlanTypeEnumSerializer._fromJsonObject(object["plan_type"]),
      status: PlanStatusEnumSerializer._fromJsonObject(object["status"]),
      version: object["version"],
    };
  },

  _toJsonObject(self: PlanEventData): any {
    return {
      created_at: self.createdAt,
      currency: self.currency,
      description: self.description,
      name: self.name,
      plan_id: PlanIdSerializer._toJsonObject(self.planId),
      plan_type: PlanTypeEnumSerializer._toJsonObject(self.planType),
      status: PlanStatusEnumSerializer._toJsonObject(self.status),
      version: self.version,
    };
  },
};
